### PR TITLE
Fix RETRO_DEVICE_ID_POINTER_PRESSED handling

### PR DESCRIPTION
**Pull request by gblues**
== DETAILS

The joypad driver was only copying the first 16 bits when executing
get_buttons(). The touchpad button is bit 19, so as a result
RETRO_DEVICE_ID_POINTER_PRESSED would never fire.

We fix this for now by copying 32 bits.

== TESTING

Will need someone to verify. I don't  have a core handy that leverages
the pointer device state functionality.

== REVIEW
@QuarkTheAwesome @twinaphex @r-type

### CODE REVIEW

File: wiiu/input/wiiu_hid.c [hunk 1]
```diff
@@ -458,16 +467,19 @@ static void wiiu_hid_polling_thread_cleanup(OSThread *thread, void *stack) {
...
-  wiiu_attach_event *event;
+  wiiu_attach_event *event, *event_next = NULL;
+
   if(!hid || !list)
     return;
 
-  for(event = list; event != NULL; event = event->next) {
+  for(event = list; event != NULL; event = event_next ) {
+    event_next  = event->next;
     if(event->type == HID_DEVICE_ATTACH) {
```
Comment (twinaphex) [inline wiiu/input/wiiu_hid.c]: Some minor style nits BTW - if a conditional block only has one line, drop the brackets around it.

**Comment by r-type** [comment]
I can't test now , i will later .

BTW here is the test pointer core , in case someone want to test before i can.

https://github.com/r-type/testpointer

**Comment by r-type** [comment]
@cucholix  this one is only for the pointer issue!

@gblues @QuarkTheAwesome  pointer pressed is not yet reported ,
however state.data[0] is well update with 0x8000.
if i change the code in https://github.com/libretro/RetroArch/blob/master/input/drivers/wiiu_input.c#L108
and replace 
```
return BIT256_GET(state, VPAD_BUTTON_TOUCH) ? 1 : 0;
```
by 
```
return (state.data[0]&VPAD_BUTTON_TOUCH) ? 1 : 0;
```
the pointer press state is reported correctly , so there maybe an issue with the BIT256_GET macro
#define BIT256_GET(a, bit)       BIT128_GET(a, bit)

**Comment by r-type** [comment]
BTW should not be 19 here  instead VPAD_BUTTON_TOUCH ?
return BIT256_GET(state, 19/*VPAD_BUTTON_TOUCH*/) ? 1 : 0;

**Comment by gblues** [comment]
This is probably an endianness issue at this point--basically, the bits aren't  in the order we're assuming they are.

**Comment by r-type** [comment]
@gblues are you sure ?

it doesn't make sense for me to use  BIT256_GET(state, VPAD_BUTTON_TOUCH)
as VPAD_BUTTON_TOUCH is 0x80000
 BIT256_GET(state,0x80000)
#define BIT256_GET(a, bit)       BIT128_GET(a, bit)
#define BIT128_GET(a, bit)   (((a).data[(bit) >> 5] >> ((bit) & 31)) & 1)
(((a).data[(0x80000) >> 5] >> ((0x80000) & 31)) & 1)
chance it does not crash here !

don't we have to ask for bit 19 instead of 0x80000 ?
as you stated that The touchpad button is bit 19
this make more sens for me 

(((a).data[(19) >> 5] >> ((19) & 31)) & 1)

**Comment by gblues** [comment]
Maybe I am using the wrong term? I am counting from 0. Here is what the `button_state` looks like

                   rl        v33p
              h-+rlzzdurlyxbatrlt
    buttons: 0000000000000000000000000000000000000000000000000000000000000000
    // bit # 0000000000111111111122222222223333333333444444444455555555556666
    //       0123456789012345678901234567890123456789012345678901234567890123

So if you're thinking of "bit 1 == 0x1" then I  should instead say that VPAD_BUTTON_TOUCH  is bit  20,  and the 0x80000 mask makes more sense.

Where I think endianness might be coming into play is that the rarch_bits_t structure is an array of uint32_t's, and endianness is causing the translation of the uint64_t onto the rarch_bits_t map to not work as  expected.

But, that is a guess shooting off the hip. I will investigate proper later. :)

**Comment by r-type** [comment]
I understand your point ,

 what i am sure (i have use debug printf ) is that when you touch screen 
 state.data[0]  is 0x80000.

 but what i don't understand is here

 https://github.com/libretro/RetroArch/blob/master/input/drivers/wiiu_input.c#L108
 it use 
 BIT256_GET(state, VPAD_BUTTON_TOUCH)

if we look at the macro def 
#define VPAD_BUTTON_TOUCH               0x00080000
#define BIT256_GET(a, bit) BIT128_GET(a, bit)
#define BIT128_GET(a, bit) (((a).data[(bit) >> 5] >> ((bit) & 31)) & 1)

so if i am correct (maybe not ) 
this lead to

(((state).data[(0x00080000) >> 5] >> ((0x00080000) & 31)) & 1)
(0x00080000) >> 5 = 4000 ? this doesn’t make sense for me state.data[4000]

from what i understand , if we query bit 19 it reside on data[0] ( 19>>5)
if we query bit 32 it reside on data[1] (32>>5 ) ect ....
So if VPAD_BUTTON_TOUCH is bit 19 (or 20) I think we should query 
BIT256_GET(state, 19) and not  BIT256_GET(state, 0X80000)

**Comment by gblues** [comment]
Oh! Yes, I think I understand what you're  saying now.

BIT256_GET(...) is expecting a 0-based bit number (19) and we are instead giving it the actual bitmask (0x80000).

**Comment by r-type** [comment]
yes that the point ! 
now I don't know how to convert 0x80000 to 19 (or 20)

**Comment by QuarkTheAwesome** [comment]
You could look at writing a macro (usual name for this sort of thing is
ffs) using GCC's __builtin_clz (count leading zeroes). Alternatively, just
change the code to 19 and leave a comment ;D

**Pull request merged**